libs/providers/in-memory/src/lib/in-memory-provider.ts

```diff
@@ -1,16 +1,18 @@
 import {
   FlagNotFoundError,
-  FlagValue,
-  GeneralError,
   JsonValue,
   OpenFeatureEventEmitter,
   Provider,
   ProviderEvents,
   ResolutionDetails,
   StandardResolutionReasons,
   TypeMismatchError,
+  Logger,
+  EvaluationContext,
+  TargetingKeyMissingError,
+  FlagValueType,
 } from '@openfeature/js-sdk';
-import { FlagConfiguration } from './flag-configuration';
+import { FlagConfiguration, Flag } from './flag-configuration';
 
 export class InMemoryProvider implements Provider {
   public readonly events = new OpenFeatureEventEmitter();
```

Comment:
```suggestion
  public readonly events = new OpenFeatureEventEmitter();
  public readonly runsOn = 'server';
```

This will add a runtime compatibility check once [v1.5.0](https://github.com/open-feature/js-sdk/pull/560) is released.